_targets/build.project.riscv64-spike

```diff
@@ -54,37 +56,51 @@ b_build_target() {
 	# Relink kernel to include newly created cpio archive
 	(cd phoenix-rtos-kernel && make $MAKEFLAGS all)
 
+	# Convert kernel to flat binary
+	${CROSS}objcopy -O binary ${PREFIX_PROG}/phoenix-${TARGET}.elf ${PREFIX_PROG}/phoenix-${TARGET}.bin
+
 	# Add kernel to boot loader
 	b_log "Adding kernel to boot loader"
 
-	mkdir -p riscv-pk/build-${TARGET}
-	pushd riscv-pk/build-${TARGET}
-		OLDCFLAGS="$CFLAGS"
-		CFLAGS=""
-
-		if [ "X${CLEAN}" == "Xclean" ]; then
-
-			#make distclean
-			../configure --host=riscv64-phoenix-elf --with-payload=../../_build/riscv64-spike/prog/phoenix-riscv64-spike.elf --disable-fp-emulation
-		fi
-
-		make ${MAKEFLAGS} all
-		CFLAGS="$OLDCFLAGS"
-	popd
-
-	cp riscv-pk/build-${TARGET}/bbl $PREFIX_BOOT/phoenix-riscv64-spike.bbl
+	if [ "${RISCV_LOADER}" == "riscv-pk" ]; then
+		mkdir -p riscv/riscv-pk/build-${TARGET}
+		pushd riscv/riscv-pk/build-${TARGET}
+			if [ "X${CLEAN}" == "Xclean" ]; then
+				#make distclean
+				../configure --host=${CROSS%?} --with-payload=${PREFIX_PROG}/phoenix-${TARGET}.elf --disable-fp-emulation
+			fi
+
+			env -i PATH="$PATH" make ${MAKEFLAGS} all
+		popd
+	elif [ "${RISCV_LOADER}" == "opensbi" ]; then
+		pushd riscv/opensbi/
+			if [ "X${CLEAN}" == "Xclean" ]; then
+				make distclean
+			fi
+
+			make CROSS_COMPILE=${CROSS} PLATFORM=generic FW_PAYLOAD_PATH=${PREFIX_PROG}/phoenix-${TARGET}.bin
+		popd
+	fi
 }
 
 
 b_image_target() {
-	b_log "Creating image from $PREFIX_ROOTFS"
+	b_log "Creating image from ${PREFIX_ROOTFS}"
 	size=`find ${PREFIX_ROOTFS} -type f -exec du -k {} \; | awk 'BEGIN { sum=0 }; { sum+=$1; }; END { print sum }'`
 	size=`expr ${size} \* 150 / 100`
-	echo "Filesystem size: $size"
-
-	genext2fs -b $size -d $PREFIX_ROOTFS $PREFIX_BUILD/disk.ext2
-	cp $PREFIX_BUILD/disk.ext2 $PREFIX_BOOT/phoenix-riscv64-spike.disk
-
+	echo "Filesystem size: ${size}"
+
+	genext2fs -b ${size} -d ${PREFIX_ROOTFS} ${PREFIX_BUILD}/disk.ext2
+	cp ${PREFIX_BUILD}/disk.ext2 ${PREFIX_BOOT}/phoenix-${TARGET}.disk
```

Comment:
why not mv? it would be much quicker

Reply:
Fixed. I agree with using mv here since disk.ext2 partition is generated in the very same function (in the previous line). However, in general I prefer to copy the final artifacts and here is why:
1. Imagine that someone has built final image with "clean all" options passed in to build.sh script.
2. The person plays with resulting image and breaks the ext2 partition.
3. Rebuilding the partition by passing "image" option to build.sh should suffice (it wouldn't if the disk was generated in other function and we would use mv instead of cp).

Reply:
I don't think I understand how `cp` instead `mv` would help in case user would break the ext2 partition. If we're considering user breaking something, he can break any file and using just `image` build target will result in unbootable image.

`mv` as opposed to `cp` is atomic (assuming operations within the same partition) so it actually reduces number of potential problems.

I think it's not all that important regarding this PR and can be addressed separately in the future.